### ISSUE TITLE
key-spacing align colon plus one-line: true

### ISSUE THREAD

**Issue by Globegitter** [eslint/eslint#2018]
I like the rule `"key-spacing": [2, { "align": "colon" }]` for most times, especially when yo create bigger objects, but it would also be great to have some option, to still allow an object to be created in one line.

So that this:
```js
spawn(cmd, cmdOptions, { cwd: clientName, stdio: 'inherit' });
```
and this
```js
var obj = {
    foobar  : 42,
    bat     : (2 * 2),
    default : fn()
};
```

would pass, but nothing else. Does that make sense? Is that in any way currently possible or could that be added?

**Comment by IanVS** [comment]
@Globegitter I can't test right now, but did you try:

    "key-spacing": [2, { "align": "colon" , "beforeColon": false, "afterColon": true }]

The docs state "Except in the case of padding, it still respects `beforeColon` and `afterColon`."

**Comment by Globegitter** [comment]
Yep, I did try. It then throws:

```
  48:35  error  Missing space after key "cwd"  key-spacing
  48:53  error  Extra space after key "stdio"  key-spacing
``

**Comment by nzakas** [comment]
Just so we are clear, can you share the code, the configuration for the rule you are using, and the output your seeing? Makes it much easier to understand what's going on.

**Comment by ldiqual** [comment]
+1 for one-line exceptions

**Comment by IanVS** [comment]
+1 from me as well.  It doesn't seem like this rule should also force a multi-line structure.

**Comment by nzakas** [comment]
+1's aren't very useful. If you'd like this feature, please make a proposal, or even better, submit a pull request. We have lots of open issues and few people helping to close them.

**Comment by IanVS** [comment]
I haven't delved into the rules before, but I'll try to take a crack at it.
